merkletree/merkletreekv.go

```diff
@@ -0,0 +1,104 @@
+package merkletree
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/storage/kv"
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+var (
+	ErrorBadTreeNonce = errors.New("[merkletree] Bad tree nonce")
+)
+
+func NewMerkleTreeFromKV(db kv.DB, epoch uint64) (*MerkleTree, error) {
+	nonceKey := append([]byte{TreeNonceIdentifier}, util.ULongToBytes(epoch)...)
+	val, err := db.Get(nonceKey)
+	var nonce []byte
+	if err != nil {
+		return nil, err
+	} else if len(val) != crypto.HashSizeByte {
+		return nil, ErrorBadTreeNonce
+	} else {
+		nonce = val
+	}
+	m := new(MerkleTree)
+	m.nonce = nonce
+	root, err := m.reconstructTree(db, nil, epoch, []bool{})
+	if err != nil {
+		return nil, err
+	}
+	m.root = root.(*interiorNode)
+	m.hash = m.root.Hash(m)
+	return m, nil
+}
+
+func (m *MerkleTree) reconstructTree(db kv.DB, parent MerkleNode, epoch uint64, prefixBits []bool) (MerkleNode, error) {
```

Comment:
Is there a reason why ``reconstructTree`` needs to be a method of ``MerkleTree``? None of ``m``'s fields are used in this function.